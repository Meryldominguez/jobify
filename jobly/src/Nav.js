import {
  Navbar,
  Nav as N,

  // Form,
  // FormControl,
  // Button
} from "react-bootstrap"
import {
  Link
} from "react-router-dom"
import './App.css';

function Nav({isLoggedin=true, logout=()=> console.log("Logged out")}) {

  return (
    <Navbar bg="light" expand="md">
      <Navbar.Brand as={Link} to="/">Jobify</Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <N className="ml-auto">
          
          {isLoggedin? 
          <>
          <N.Link as={Link} to="/companies">Companies</N.Link>
          <N.Link as={Link} to="/jobs">Jobs</N.Link>
          <N.Link as={Link} to="/profile">Profile</N.Link>
          <N.Link onClick={logout} >Logout</N.Link>
          </>
          :
          <>
          <N.Link as={Link} to="/login">Login</N.Link>
          <N.Link as={Link} to="/signup">Signup</N.Link>
          </>
          }
          
        </N>
        {/* <Form inline>
          <FormControl type="text" placeholder="Search" className="mr-sm-2" />
          <Button variant="outline-success">Search</Button>
        </Form> */}
      </Navbar.Collapse>
    </Navbar>
  );
}

export default Nav;

