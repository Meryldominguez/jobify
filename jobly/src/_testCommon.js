import { unmountComponentAtNode } from "react-dom";

let container

const commonBeforeEach= () => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container)
};

const commonAfterEach= () => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container= null
};

export {commonBeforeEach,commonAfterEach}