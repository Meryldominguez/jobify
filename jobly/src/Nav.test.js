import { render  } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import UserContext from './context/UserContext';
import Nav from './Nav';
import { commonBeforeEach, commonAfterEach } from "./_testCommon";

beforeEach(commonBeforeEach)
afterEach(commonAfterEach)

const customRender = (ui, { providerProps, ...renderOptions }) => {
  return render(
    <UserContext.Provider value={{...providerProps}}>
      <MemoryRouter>
      {ui}
      </MemoryRouter>
      </UserContext.Provider>,
    renderOptions
  )
}
describe("smoke tests, and testing for user", ()=>{
  test('Smoke test', () => {
    customRender(<Nav  />, {user:{}, isLoading:false});
  });
  test('Snapshot', () => {
   const {asFragment}=customRender(<Nav  />, {user:{}, isLoading:false});
   const noUser = asFragment()
   expect(noUser).toMatchSnapshot()
  });

  test('Rendering with no user', () => {
    const {getByText} = customRender(<Nav  />, {user:{}, isLoading:false});
    expect(getByText("Jobify")).toBeInTheDocument()

    expect(getByText("Login")).toBeInTheDocument()
    expect(getByText("Signup")).toBeInTheDocument()

  });

  test('Navbar responds to user', () => {
    const providerProps = {
      user:{username:"Testuser"},
      isLoading:false
    }
    const {getByText} = customRender(<Nav />, { providerProps })
    expect(getByText("Jobify")).toBeInTheDocument()

    expect(getByText("Companies")).toBeInTheDocument()
    expect(getByText("Jobs")).toBeInTheDocument()
    expect(getByText("Profile")).toBeInTheDocument()
    expect(getByText("Logout")).toBeInTheDocument()
  })
})


// describe("navigation and router tests", ()=>{
//   test("logging out",()=>{
//     let providerProps = {
//       user:{username:"Testuser"},
//       isLoading:false
//     }

//     const {debug, getByText,asFragment} = customRender(<Nav />, {providerProps});
//     const firstRender = asFragment()
//     expect(firstRender).toMatchSnapshot()

//     expect(getByText("Logout")).toBeInTheDocument()

//     fireEvent.click(getByText("Logout"))
//     debug()

   
//   })
// })