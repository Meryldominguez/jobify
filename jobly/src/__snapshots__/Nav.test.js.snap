// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke tests, and testing for user Snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-md navbar-light bg-light"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Jobify
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        class="ml-auto navbar-nav"
      >
        <a
          class="nav-link"
          href="/login"
        >
          Login
        </a>
        <a
          class="nav-link"
          href="/signup"
        >
          Signup
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
